clang/lib/CConv/AVarBoundsInfo.cpp

```diff
@@ -662,183 +640,320 @@ AvarBoundsInference::
     }
     AllVars.clear();
     if (BVar)
-      AllVars.insert(BVar);
+      AllVars.insert(BVar->getKey());
   }
 }
 
-bool AvarBoundsInference::inferPossibleBounds(BoundsKey K, ABounds *SB,
-                                              AVarGraph &BKGraph,
-                                              std::set<ABounds *> &EB) {
-  bool RetVal = false;
-  if (SB != nullptr) {
-    auto *Kvar = BI->getProgramVar(K);
-    bool ValidB = false;
-    auto BKind = SB->getKind();
-    BoundsKey SBKey;
-    if (CountBound *CB = dyn_cast<CountBound>(SB)) {
-      ValidB = true;
-      SBKey = CB->getCountVar();
-    } else if (ByteBound *BB = dyn_cast<ByteBound>(SB)) {
-      ValidB = true;
-      SBKey = BB->getByteVar();
-    }
-
-    std::set<ProgramVar *> PotentialB;
-    // If we can handle the bounds?
-    if (ValidB) {
-      // First, find all the in-scope variable to which the SBKey flow to.
-      auto *SBVar = BI->getProgramVar(SBKey);
-      if (SBVar->IsNumConstant()) {
-        PotentialB.insert(SBVar);
+bool
+AvarBoundsInference::convergeInferredBounds() {
+  bool FoundSome = false;
+  for (auto &InfABnds : CurrIterInferBounds) {
```

Comment:
Where does `InfABnds` get set? I didn't notice it in the `performXXX` functions.

Reply:
It gets set in `predictBounds` function